lib/ansible/galaxy/collection/__init__.py

```diff
@@ -246,6 +263,26 @@ def verify_local_collection(
     return result
 
 
+def verify_file_signature(b_manifest_file, b_detached_signature, keyring):
+    gpg_result, rc = run_gpg_verify(b_manifest_file, b_detached_signature, keyring, display)
+
+    errors = None
+    if gpg_result:
+        errors = parse_gpg_errors(gpg_result)
+
+    if errors:
+        verify_failed = "Signature verification failed:"
+        reasons = set(error.explain() for error in errors)
+        for reason in reasons:
+            verify_failed += f'\n      * {reason}'
+        display.display(verify_failed)
+        return False
+    if rc:
+        display.display(f"Unexpected error: GnuPG signature verification failed with the return code {rc} and output {gpg_result}")
+        return False
+    return True
```

Comment:
how about
```suggestion
    return not rc
```

Reply:
With the patch above this would be:
```suggestion
    return not gpg_verification_failed
```